src/mitol/payment_gateway/api.py

```diff
@@ -306,3 +375,90 @@ def perform_processor_response_validation(self, request):
         signature = self._generate_cybersource_sa_signature(passed_payload)
 
         return passed_payload["signature"] == signature
+
+    def convert_to_order(response):
+        """
+        CyberSource includes the order in its response, using the same
+        field names prepended with "req_". This will grab those fields
+        out of the response and convert it back to an Order and a set of
+        CartItems. (Not all payment processor APIs may support this, so this
+        is just defined for the CyberSource one.)
+
+        Args:
+            response: Dict; the data from the response
+
+        Returns:
+            Order
+        """
+        items = []
+
+        for i in range(0, int(response["req_line_item_count"])):
+            line = CartItem(
+                code=response[f"req_item_{i}_code"],
+                name=response[f"req_item_{i}_name"],
+                sku=response[f"req_item_{i}_sku"],
+                unitprice=response[f"req_item_{i}_unit_price"],
+                taxable=response[f"req_item_{i}_tax_amount"],
+                quantity=response[f"req_item_{i}_quantity"],
+            )
+            items.append(line)
+
+        order = Order(
+            username=response["req_consumer_id"],
+            ip_address=response["req_customer_ip_address"],
+            reference=response["req_reference_number"],
+            items=items,
+        )
+
+        return order
+
+    def decode_processor_response(self, request):
+        """
+        The CyberSource implementation of this. The ProcessorResponse
+        class was designed around this processor so the state codes map
+        one-to-one to the decision field that gets passed back from them.
+
+        The reason code and transaction ID fields don't appear in certain
+        states.
+        """
+        if request.method == "POST":
+            if hasattr(request, "data"):
+                response = request.data
+            else:
+                response = request.POST
+        else:
+            if hasattr(request, "query_params"):
+                response = request.query_params
+            else:
+                response = request.GET
```

Comment:
Can simplify these down using the `getattr` builtin:
```suggestion
            getattr(request, "query_params", request.GET)
```